### DISCUSSION TITLE
how to split one record in two or more in MV

### DISCUSSION THREAD

**Discussion by vpease** [ClickHouse/ClickHouse#34926] [Q&A]
I have a Kafka source with network flows information. Every message is a flow where there is a Datasets property which is an array. 
What I need its to split the record coming from Kafka in as much records as elements in the array. 
The message in Kafka is like this:
{"AgentID":"192.168.1.250","Header":{"Version":9,"Count":2},"DataSets":[
 [{"I":8,"V":"192.168.10.99"},{"I":12,"V":"200.48.225.130"},{"I":15,"V":"192.168.1.1"},{"I":10,"V":"0x0010"},{"I":14,"V":"0x000b"},{"I":2,"V":"0x00000001"},{"I":1,"V":"0x00000049"},{"I":22,"V":2746593},{"I":21,"V":2746593},{"I":7,"V":59803},{"I":11,"V":53},{"I":6,"V":"0x00"},{"I":4,"V":17},{"I":5,"V":0},{"I":16,"V":"0x0000"},{"I":17,"V":"0x0000"},{"I":9,"V":24},{"I":13,"V":0}],

 [{"I":8,"V":"192.168.10.99"},{"I":12,"V":"200.48.225.130"},{"I":15,"V":"192.168.1.1"},{"I":10,"V":"0x0010"},{"I":14,"V":"0x000b"},{"I":2,"V":"0x00000001"},{"I":1,"V":"0x00000049"},{"I":22,"V":2746593},{"I":21,"V":2746593},{"I":7,"V":59803},{"I":11,"V":53},{"I":6,"V":"0x00"},{"I":4,"V":17},{"I":5,"V":0},{"I":16,"V":"0x0000"},{"I":17,"V":"0x0000"},{"I":9,"V":24},{"I":13,"V":0}]
]}

Using the Kafka connector and a MV that will be one record but I need it as two records. One per every element in the Dataset array.

How can I do that?

For now, my solution is to include a Kafka client app in NodeJS doing this before inserting in CH.

**Comment by UnamedRus** [comment]
JSONAsString format in Kafka table + JSONExtract + ARRAY JOIN in MV

https://kb.altinity.com/altinity-kb-queries-and-syntax/jsonextract-to-parse-many-attributes-at-a-time/
https://kb.altinity.com/altinity-kb-schema-design/altinity-kb-jsonasstring-and-mat.-view-as-json-parser/
